src/pages/RFOX/hooks/useCurrentEpochRewardsQuery.ts

```diff
@@ -1,82 +1,90 @@
 import type { UseQueryResult } from '@tanstack/react-query'
-import { skipToken, useQueries } from '@tanstack/react-query'
+import { useQueries } from '@tanstack/react-query'
 import { useCallback } from 'react'
 import { mergeQueryOutputs } from 'react-queries/helpers'
+import { getAddress } from 'viem'
 
-import type { PartialEpochMetadata } from '../types'
+import type { CurrentEpochMetadata, Epoch } from '../types'
 import { calcEpochRewardForAccountRuneBaseUnit } from './helpers'
-import { useCurrentEpochMetadataQuery } from './useCurrentEpochMetadataQuery'
+import { getAffiliateRevenueQueryFn, getAffiliateRevenueQueryKey } from './useAffiliateRevenueQuery'
 import { getEarnedQueryFn, getEarnedQueryKey } from './useEarnedQuery'
+import { fetchEpochHistory, getEpochHistoryQueryKey } from './useEpochHistoryQuery'
 
 type UseCurrentEpochRewardsQueryProps = {
   stakingAssetAccountAddress: string | undefined
+  currentEpochMetadata: CurrentEpochMetadata | undefined
 }
 
 /**
  * Gets the rewards so far for the current epoch for a given account address.
  */
 export const useCurrentEpochRewardsQuery = ({
   stakingAssetAccountAddress,
+  currentEpochMetadata,
 }: UseCurrentEpochRewardsQueryProps) => {
-  const currentEpochMetadataQuery = useCurrentEpochMetadataQuery()
-
   const combine = useCallback(
-    (_queries: [UseQueryResult<bigint, Error>, UseQueryResult<bigint, Error>]) => {
-      const combineResults = (_results: (bigint | PartialEpochMetadata)[]) => {
-        const results = _results as [bigint, bigint, PartialEpochMetadata]
-        const currentEpochMetadata = currentEpochMetadataQuery.data
-        const [previousEpochEarned, currentEpochEarned] = results
+    (
+      queries: [
+        UseQueryResult<Epoch[], Error>,
+        UseQueryResult<bigint, Error>,
+        UseQueryResult<bigint, Error>,
+      ],
+    ) => {
+      const combineResults = (
+        _results: (Epoch[] | bigint | CurrentEpochMetadata | undefined)[],
+      ) => {
+        if (!stakingAssetAccountAddress) return 0n
 
-        const epochEarningsForAccount = currentEpochEarned - previousEpochEarned
+        const results = _results as [Epoch[] | undefined, bigint | undefined, bigint | undefined]
```

Comment:
Suggestion: We should probably use named tuple types here to makes this more readable.

```ts
type EpochRewardsResultTuple = [
  epochHistory: Epoch[] | undefined,
  currentEpochRewardUnits: bigint | undefined,
  affiliateRevenue: bigint | undefined
];

const results = _results as ResultTuple;
```

Reply:
https://github.com/shapeshift/web/pull/7366/commits/c21b0187f74d9907f8778932de2bebd5d05a7837